src/main/java/org/spongepowered/api/text/selector/Argument.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.text.selector;
+
+/**
+ * Represents the argument of a {@link Selector selector}. An Argument is one or
+ * more key-to-value mappings ({@code key=value}) with the key represented by an
+ * {@link ArgumentType}.
+ *
+ * @param <T> The type of the value
+ */
+public interface Argument<T> {
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  adam@eq2.serveftp.net
    Domain eq2.co.uk has exceeded the max emails per hour (60/60 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o4.sgmail.github.com ([192.254.112.99]:58491)
	by web3.cov.yesit.co.uk with esmtps (TLSv1.2:DHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.84)
	(envelope-from <bounces+848413-d0b0-adam=eq2.co.uk@sgmail.github.com>)
	id 1YNeXz-0003Ty-7y
	for adam@eq2.co.uk; Tue, 17 Feb 2015 09:34:27 +0000
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed; d=github.com; 
	h=from:reply-to:to:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=3mUqMzLBasz9mDnYvrexU/TBt6E=; b=o2EreRGquZ8DBKSn
	6o4OE83W/qdwSjy3U6M1K/GAfLASo47DBk5/uqU17Hv57NrmoGMB55DAdFFZJIap
	RSqCX+ossrNS4A/B6UJD7qNELB5cUOz+tPZQRvzZzyOSvnoe96b9FRV5Vecmh5IH
	55K+uy+GQF3j6Xzv4eRk/k/iMdw=
Received: by filter0296p1mdw1.sendgrid.net with SMTP id filter0296p1mdw1.30657.54E30B2016
        2015-02-17 09:34:25.818723859 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd-031 (SG) with ESMTP id 14b96e37bbc.3c4b.25cda8
	for <adam@eq2.co.uk>; Tue, 17 Feb 2015 09:34:25 +0000 (UTC)
Date: Tue, 17 Feb 2015 01:34:25 -0800
From: gratimax <notifications@github.com>
Reply-To: SpongePowered/SpongeAPI <reply@reply.github.com>
To: SpongePowered/SpongeAPI <SpongeAPI@noreply.github.com>
Message-ID: <SpongePowered/SpongeAPI/pull/425/r24802993@github.com>
In-Reply-To: <SpongePowered/SpongeAPI/pull/425@github.com>
References: <SpongePowered/SpongeAPI/pull/425@github.com>
Subject: Re: [SpongeAPI] [WIP] Text API 2 (#425)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_54e30b2193b3e_3af23fd2ed5352a0623696";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: gratimax
X-GitHub-Recipient: Mumfrey
List-ID: SpongePowered/SpongeAPI <SpongeAPI.SpongePowered.github.com>
List-Archive: https://github.com/SpongePowered/SpongeAPI
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+001180bfcdd42e78d07ab90b03dc742a985f5a767aa909d592cf0000000110facd2192a170ce017a76b1@reply.github.com>,
 <https://github.com/notifications/unsubscribe/ABGAvzXd5WcZCWtAN168MvQr2-jPaT2Nks5nswKhgaJpZM4DZr_C>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: adam@eq2.co.uk
X-SG-EID: VzkE0stiinYMPYWPWpnZ4OvH/Ru8MrIuYzb7JqhnFX+dNKYUd8M5UxVx/pjvivQ30alP/byAbbKjdD
 Zs/KUnuPiybcJLbw/zmfO0TL5y6eYzEW9hWxpa4ozidrjgKcDCH7buNMqHU7cP8k6/cGlt5U/gnV8P
 pdhajmMlgDSuH28=

----==_mimepart_54e30b2193b3e_3af23fd2ed5352a0623696
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

> + * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
> + * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
> + * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
> + * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
> + * THE SOFTWARE.
> + */
> +package org.spongepowered.api.text.selector;
> +
> +/**
> + * Represents the argument of a {@link Selector selector}. An Argument is one or
> + * more key-to-value mappings ({@code key=value}) with the key represented by an
> + * {@link ArgumentType}.
> + *
> + * @param <T> The type of the value
> + */
> +public interface Argument<T> {

This seems like a fantastic candidate to be turned into a class

---
Reply to this email directly or view it on GitHub:
https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24802993
----==_mimepart_54e30b2193b3e_3af23fd2ed5352a0623696
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/SpongePowered/SpongeAPI/pull/425#discussion_r24802993">src/main/java/org/spongepowered/api/text/selector/Argument.java</a>:</p>
<pre style='color:#555'>&gt; + * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
&gt; + * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
&gt; + * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
&gt; + * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
&gt; + * THE SOFTWARE.
&gt; + */
&gt; +package org.spongepowered.api.text.selector;
&gt; +
&gt; +/**
&gt; + * Represents the argument of a {@link Selector selector}. An Argument is one or
&gt; + * more key-to-value mappings ({@code key=value}) with the key represented by an
&gt; + * {@link ArgumentType}.
&gt; + *
&gt; + * @param &lt;T&gt; The type of the value
&gt; + */
&gt; +public interface Argument&lt;T&gt; {
</pre>
<p>This seems like a fantastic candidate to be turned into a class</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24802993">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/ABGAv5HfxO5F32ES2ND07AGNkJqFSYocks5nswKhgaJpZM4DZr_C.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
  <div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
    <link itemprop="url" href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24802993"></link>
    <meta itemprop="name" content="View Pull Request"></meta>
  </div>
  <meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_54e30b2193b3e_3af23fd2ed5352a0623696--